File: frontend/src/services/reviewService.ts
import apiClient from './api';
import { ApiResponse, PageResponse } from '../types/api.types';

export interface ReviewItem {
  id: number;
  userId: number;
  userName: string;
  rating: number;
  comment: string;
  createdAt: string;
}

export const reviewService = {
  async getCarReviews(carId: number, page = 0, size = 5): Promise<PageResponse<ReviewItem>> {
    const response = await apiClient.get<ApiResponse<PageResponse<ReviewItem>>>(`/cars/${carId}/reviews`, { params: { page, size } });
    return response.data.data;
  },
  async createReview(bookingId: number, rating: number, comment: string): Promise<ReviewItem> {
    const response = await apiClient.post<ApiResponse<ReviewItem>>('/reviews', { bookingId, rating, comment });
    return response.data.data;
  },
};

export default reviewService;

File: backend/src/main/java/com/driveease/service/ReviewService.java
package com.driveease.service;

import com.driveease.dto.request.ReviewCreateRequest;
import com.driveease.dto.response.PageResponse;
import com.driveease.dto.response.ReviewResponse;

public interface ReviewService {
    PageResponse<ReviewResponse> getCarReviews(Long carId, int page, int size);
    ReviewResponse createReview(Long userId, ReviewCreateRequest request);
}
